### PR TITLE
Support XGROUP CREATECONSUMER

### LINKED ISSUE #2905: Add support for redis command: XGROUP CREATECONSUMER

### PR DESCRIPTION
**Pull request by Avital-Fine**
Closes #2905

### CODE REVIEW

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 1]
```diff
@@ -2273,6 +2273,11 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
         .key(key).add(groupname), BuilderFactory.LONG);
   }
 
+  public final CommandObject<Long> xgroupCreateConsumer(String key, String groupname, String consumername) {
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/CommandObjects.java:2276]:
```suggestion
  public final CommandObject<Long> xgroupCreateConsumer(String key, String groupName, String consumerName) {
```

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 2]
```diff
@@ -2295,11 +2300,15 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
...
 
-  public final CommandObject<Long> xgroupDelConsumer(byte[] key, byte[] groupname, byte[] consumerName) {
-    return new CommandObject<>(commandArguments(XGROUP).add(DELCONSUMER)
-        .key(key).add(groupname).add(consumerName), BuilderFactory.LONG);
+  public final CommandObject<Long> xgroupCreateConsumer(byte[] key, byte[] groupname, byte[] consumername) {
+    return new CommandObject<>(commandArguments(XGROUP).add(CREATECONSUMER)
+        .key(key).add(groupname).add(consumername), BuilderFactory.LONG);
   }
 
+  public final CommandObject<Long> xgroupDelConsumer(byte[] key, byte[] groupname, byte[] consumername) {
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/CommandObjects.java:2308]:
```suggestion
  public final CommandObject<Long> xgroupDelConsumer(byte[] key, byte[] groupName, byte[] consumerName) {
```

File: src/main/java/redis/clients/jedis/commands/StreamBinaryCommands.java [hunk 3]
```diff
@@ -31,6 +31,8 @@
 
   long xgroupDestroy(byte[] key, byte[] groupname);
 
+  long xgroupCreateConsumer(byte[] key, byte[] groupname, byte[] consumerName);
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/commands/StreamBinaryCommands.java:34]: add javadoc

File: src/main/java/redis/clients/jedis/commands/StreamCommands.java [hunk 4]
```diff
@@ -128,13 +128,21 @@ default StreamEntryID xadd(String key, Map<String, String> hash, XAddParams para
    */
   long xgroupDestroy(String key, String groupname);
 
+  /**
+   * XGROUP CREATECONSUMER <key> <groupname> <consumername>
+   * @param key
+   * @param groupname
+   * @param consumername
+   */
+  long xgroupCreateConsumer(String key, String groupname, String consumername);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/StreamCommands.java:137]:
```suggestion
  boolean xgroupCreateConsumer(String key, String groupname, String consumername);
```

File: src/test/java/redis/clients/jedis/commands/jedis/StreamsCommandsTest.java [hunk 5]
```diff
@@ -369,13 +369,11 @@ public void xgroup() {
...
 
     assertEquals("OK", jedis.xgroupCreate("xgroup-stream", "consumer-group-name", null, false));
-
     assertEquals("OK", jedis.xgroupSetID("xgroup-stream", "consumer-group-name", id1));
-
     assertEquals("OK", jedis.xgroupCreate("xgroup-stream", "consumer-group-name1", StreamEntryID.LAST_ENTRY, false));
 
     jedis.xgroupDestroy("xgroup-stream", "consumer-group-name");
-
+    assertEquals(1L, jedis.xgroupCreateConsumer("xgroup-stream", "consumer-group-name1","myconsumer1"));
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/commands/jedis/StreamsCommandsTest.java:376]: Do this after following assert. Also try to delete that consumer (again). You can also consider using a different groupname.

File: src/main/java/redis/clients/jedis/Protocol.java [hunk 6]
```diff
@@ -261,7 +261,7 @@ private Command() {
     TIMEOUT, ABORT, NX, XX, EX, PX, EXAT, PXAT, CH, WITHCOORD, WITHDIST, WITHHASH, STOREDIST, COPY,
     KEEPTTL, AUTH, AUTH2, INFO, CHANNELS, NUMPAT, NUMSUB, FULL, NOW, VERSION, KEYS, IDX, SCHEDULE,
     ANY, FROMMEMBER, FROMLONLAT, BYRADIUS, BYBOX, BYLEX, BYSCORE, REV, MINMATCHLEN, WITHMATCHLEN,
-    PURGE, STATS, @Deprecated LCS, @Deprecated STRINGS;
+    PURGE, STATS, @Deprecated LCS, @Deprecated STRINGS, CREATECONSUMER;
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Protocol.java:264]:
```suggestion
    PURGE, STATS, CREATECONSUMER, @Deprecated LCS, @Deprecated STRINGS;
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**